### DISCUSSION TITLE
Using git clone and Zip download

### DISCUSSION THREAD

**Discussion by Werterror** [community/community#23121] [New to GitHub]
<p>Hi all,</p>
<p>I would like to download a repository related to this tutorial:</p><aside class="onebox githubblob" data-onebox-src="https://github.com/Xilinx/Vitis-AI-Tutorials/blob/3549b06189db3e553480587286b384cd25912f76/Design_Tutorials/09-mnist_pyt/README.md">
  <header class="source">

      <a href="https://github.com/Xilinx/Vitis-AI-Tutorials/blob/3549b06189db3e553480587286b384cd25912f76/Design_Tutorials/09-mnist_pyt/README.md" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>

  <article class="onebox-body">
    <h4><a href="https://github.com/Xilinx/Vitis-AI-Tutorials/blob/3549b06189db3e553480587286b384cd25912f76/Design_Tutorials/09-mnist_pyt/README.md" target="_blank" rel="noopener nofollow ugc">Xilinx/Vitis-AI-Tutorials/blob/3549b06189db3e553480587286b384cd25912f76/Design_Tutorials/09-mnist_pyt/README.md</a></h4>

      <pre><code class="lang-md">&lt;!--
Copyright 2020-2022 Xilinx Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Author: Mark Harvey, Xilinx Inc
--&gt;
&lt;table class="sphinxhide"&gt;
 &lt;tr&gt;
   &lt;td align="center"&gt;&lt;img src="https://raw.githubusercontent.com/Xilinx/Image-Collateral/main/xilinx-logo.png" width="30%"/&gt;&lt;h1&gt;Vitis AI Tutorials&lt;/h1&gt;
</code></pre>

  This file has been truncated. <a href="https://github.com/Xilinx/Vitis-AI-Tutorials/blob/3549b06189db3e553480587286b384cd25912f76/Design_Tutorials/09-mnist_pyt/README.md" target="_blank" rel="noopener nofollow ugc">show original</a>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
<p>
In order to do so I navigate to Xililnx/Vitis-AI-Tutorials, copy http adress and use git clone to clone it to desired folder.<br>
After that in my folder I can find 1.37 GB of files but there is nothing that I want.<br>
But when I use “Download ZIP” I get less then 500MB of files that I expected to get cloned.<br>
Shouldn’t those to operations bring same results?<br>
I am still learning about features of github, so please understand my confiusion.</p>
<p>Best regards,</p>

**Comment by jfullstackdev** [comment]
<p>it’s just README, on my end when I downloaded it, it’s just 4KB,</p>
<p>maybe your question is about downloading and cloning</p>
<p><a href="https://stackoverflow.com/questions/31977306/difference-between-cloning-and-downloading#:~:text=When%20you%20download%20the%20repo,is%20a%20functional%20git%20repo" class="inline-onebox">github - Difference between cloning and downloading - Stack Overflow</a>.</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="Werterror" data-post="1" data-topic="256919">
<div class="title">
<div class="quote-controls"></div>
 Werterror:</div>
<blockquote>
<p>After that in my folder I can find 1.37 GB of files but there is nothing that I want.<br>
But when I use “Download ZIP” I get less then 500MB of files that I expected to get cloned.<br>
Shouldn’t those to operations bring same results?</p>
</blockquote>
</aside>
<p>No, though depending on a lot of factors the results may be <em>similar</em>.</p>
<ul>
<li>Clone gives you a full copy of the repository, all its history, all branches (unless you use specific options to limit that).</li>
<li>The files you see after a successful clone are those of the default branch, in case of that repository called <code>master</code>. If you want to see another branch, tag, or random commit you need to use <code>git checkout</code> (but that works without more network requests!).</li>
<li>When you download a ZIP archive you get the files of the specific repository state related to the link you had – which might be any branch, tag, or commit.</li>
</ul>
<p>If you want the state with the README you linked above, running</p>
<pre><code class="lang-auto">git checkout 3549b06189db3e553480587286b384cd25912f76
</code></pre>
<p>in the cloned repository should work. The 35… is the commit ID from your link.</p>
<p>Keep in mind though that if you want to work with the repository from there you should probably create a branch. And (depending on what you’re working on) you might want to check out an existing branch or tag instead of that specific commit.</p>
<p>Finally, if you’re not that familiar with Git yet and some of those terms are confusing, I recommend you look at the <a href="https://git-scm.com/book/en/v2" class="inline-onebox">Git - Book</a> (it’s a website despite the name). Chapters 1 through 3 explain the basics. :slightly_smiling_face:</p>

**Comment by Werterror** [comment]
<p>Thank you <a class="mention" href="/airtower-luna">@airtower-luna</a>, right on the point and clear enough that even a newbie can understand it.<br>
Also thank you for the “so called book”, it might come handy.</p>

**Comment by Werterror** [comment]
<p>I did link folder with just readme file, becouse thats the tutorial I am refering too, if you would like to navigate to upper folder, you would find whole repository.</p>
